### ISSUE TITLE
Linter not working

### ISSUE THREAD

**Issue by cideM** [mawww/kakoune#2314]
I am playing around with Kakoune and tried to make `eslint` work. As a frontend developer, using a globally installed `eslint` is just not an option. So I wrote a small bash script which finds and calls the local `eslint` and runs the output through `sed` to generate something Kakoune can understand.

Here's the output from manually calling `eslint`

```shell
❯ ../../node_modules/eslint/bin/eslint.js index.js --format compact
/Users/yuuki/private/teems/source/lib/index.js: line 27, col 63, Error - Delete `;` (prettier/prettier)
/Users/yuuki/private/teems/source/lib/index.js: line 30, col 46, Error - Delete `;` (prettier/prettier)
```

Here's the output from the script

```shell
❯ eslint_kak.sh index.js
/Users/yuuki/private/teems/source/lib/index.js:27:63: Error: Delete `;` (prettier/prettier)
/Users/yuuki/private/teems/source/lib/index.js:30:46: Error: Delete `;` (prettier/prettier)
```

Yet in Kakoune itself the file returns the following lint output
```
index.js:
index.js:66:49: Error: Parsing error: Unexpected token =>
```

Which still seems like a valid `eslint` output but I just don't really know where it's coming from.

Here's the script (minus the part that finds the local `eslint`)

```shell
#!/bin/bash
root=$(find_up.sh node_modules)
eslint_bin="${root}/node_modules/.bin/eslint"
$eslint_bin --format compact $1 \
        | sed 's|\sline\s||' \
        | sed 's|,\scol\s|:|' \
        | sed -E 's|,\s(\w+)\s-\s|: \1: |' \
        | ghead -n -1
```

My assumption was that it would be called with just one argument, hence the `$1` but I'm guessing something about that assumption is wrong.

I am very bad at bash by the way so have mercy :P

**Comment by cideM** [comment]
Ok I was able to eventually solve this by myself. Kakoune creates temporary files which are then sent to the `lintcmd`. The version I was using did not have file endings but that was apparently solved in a recent commit. Anyways, that whole temporary folder thing proved problematic because it required me to manually look for not only the local `eslint` but also the local configuration. As an aside: the example which uses the global `eslint` is cute but unusable. No front end developer will use this and frankly it's only viable for people with very limited needs when it comes to JS linting. A global `eslint` does not use locally installed plugins, which kind of defeats the purpose of a global `eslint`.

Anyhow, I took some cues from other issues and calling the `lintcmd` with `$kak_buffile` does the trick.
